src/Mvc/Mvc.ViewFeatures/src/HtmlHelperComponentExtensions.cs

```diff
@@ -67,6 +67,38 @@ public static class HtmlHelperComponentExtensions
             };
         }
 
+        /// <summary>
+        /// Renders the <paramref name="type"/> <see cref="IComponent"/>.
+        /// </summary>
+        /// <param name="htmlHelper">The <see cref="IHtmlHelper"/>.</param>
+        /// <param name="type">The component type.</param>
+        /// <param name="parameters">An <see cref="object"/> containing the parameters to pass
+        /// to the component.</param>
+        /// <param name="renderMode">The <see cref="RenderMode"/> for the component.</param>
+        /// <returns>The HTML produced by the rendered <paramref name="type"/>.</returns>
+        public static async Task<IHtmlContent> RenderComponentAsync(
+            this IHtmlHelper htmlHelper,
+            Type type,
+            RenderMode renderMode,
+            object parameters)
+        {
+            if (htmlHelper == null)
+            {
+                throw new ArgumentNullException(nameof(htmlHelper));
+            }
+
```

Comment:
We should have a null check here for type

Reply:
What's the behaviour here if you pass in a non-component type. Does it blow up with a sensible error or should we add a check?